### ISSUE TITLE
Lack of error codes for problem-report in RFC 36

### ISSUE THREAD

**Issue by lovesh** [hyperledger/aries-rfcs#256]
RFC 36 does not specify error-codes for all possible error cases like if the issuer sent undesired credential or invalid credential, etc but says 
```
Easy-to-anticipate errors reset the flow as shown in the diagrams, and use the code issuance-abandoned; more exotic errors (e.g., server crashed at Issuer headquarters in the middle of a workflow) may have different codes but still cause the flow to be abandoned in the same way.
```
Is it really ok to have `issuance-abandoned` code for all such errors or am i missing something? I know i can use the `explain` feild.

**Comment by dhh1128** [comment]
I wrote the sentence you are highlighting. When I wrote it, I wasn't happy about it, because I thought it was not enough detail--but I didn't have better info to add, and I was under a deadline, so I got lazy.

I think getting more specific would improve the robustness of this protocol in important ways. Some obvious error conditions to contemplate include:

* The terms of service you (either issuer or holder) are proposing are unacceptable.
* The price that you (the issuer) are demanding is unacceptable.
* The proof that you (the issuer) asked me to provide, as a condition of issuance, isn't something I want to provide.
* The proof that you (the holder) just provided is inadequate, so I (the issuer) can't issue the credential to you.
* Timed out
* The preview data has an error in it.
* The credential format you're requesting or proposing doesn't work for me.
* You issued something different from what you said you were going to give me.

Many of these errors are either part of another protocol (the ones about payment are an example), or they assume situations that are impossible today because the protocol isn't complex enough right now to let them happen (e.g., today we haven't clarified how we would ask for proof or terms and conditions as a prerequisite of issuance). However, brainstorming problems worth calling out in the protocol is still probably a worthwhile exercise.
